### ISSUE TITLE
Problem with Shuffling

### ISSUE THREAD

**Issue by AlexandrosEmvoliadis** [danieltan07/dagmm#28]
Isn't it kind of cheating that the samples are being randomly shuffled for testing?
It is possible that known samples (during training) are also contained on the test set.

**Comment by chiachen-chang** [comment]
Yes, it's suitable to use the KDD dataset for unsupervised learning tasks. The training set includes only positive instances, while the testing set consists of both positive and negative ones. It's important to note that all the positive instances in the test set should be included in the training set, making this set-up appropriate for unsupervised learning.
